baselines/flwr_baselines/publications/adaptive_federated_optimization/README.md

```diff
@@ -0,0 +1,68 @@
+# Adaptive Federated Optimization 
+
+The following baseline replicates the experiments in *Adaptive Federated Optimization*, which explores the use of various server-side optimizers for different datasets.
+
+**Paper Abstract:** 
+
+<center>
+<i>Federated learning is a distributed machine learning paradigm in which a large
+number of clients coordinate with a central server to learn a model without sharing
+their own training data. Standard federated optimization methods such as Federated Averaging (FEDAVG) are often difficult to tune and exhibit unfavorable
+convergence behavior. In non-federated settings, adaptive optimization methods
+have had notable success in combating such issues. In this work, we propose federated versions of adaptive optimizers, including ADAGRAD, ADAM, and YOGI,
+and analyze their convergence in the presence of heterogeneous data for general
+nonconvex settings. Our results highlight the interplay between client heterogeneity
+and communication efficiency. We also perform extensive experiments on these
+methods and show that the use of adaptive optimizers can significantly improve the
+performance of federated learning.</i>
+</center>
+
+**Paper Authors:** 
+
+Sashank J. Reddi, Zachary Charles, Manzil Zaheer, Zachary Garrett, Keith Rush,Jakub Konecný, Sanjiv Kumar, H. Brendan McMahan.
+
+
+Note: If you use this implementation in your work, please remember to cite the original authors of the paper. 
+
+**[Link to paper.](https://arxiv.org/pdf/2003.00295.pdf)**
+
+## Currently implemented
+
+We have currently implemented the following experiments: 
+
+| Dataset      | Model | Strategies
+| ----------- | ----------- | ----------- |
+| CIFAR10      | ResNet18       | FeAvg, FedYogi, FedAdam, FedAdagrad |
+| CIFAR100   | ResNet18        | FeAvg, FedYogi, FedAdam, FedAdagrad |
+
+More experiments to be added soon. 
+
+## Running experiments
+
+Experiments are organized by dataset under the `conf` folder using [hydra](https://hydra.cc/docs/tutorials/), e.g. `conf/cifar10` for the CIFAR10 dataset. 
+Each dataset contains a `config.yaml` file containing common setup variables and a `strategy` subfolder with different strategy parameters.
+
+You can run specific experiments by passing a `conf` subfolder and a given `strategy` as follows. 
+```sh
+python main.py --config-path conf/cifar10 strategy=fedyogi
```

Comment:
After followiing the instructions in `flower/baselines/README.md`, I'm unable to run this:

```
➜  baselines poetry install
Installing dependencies from lock file

No dependencies to install or update

Installing the current project: flwr_baselines (0.17.0)
➜  baselines cd flwr_baselines/publications/adaptive_federated_optimization 
➜  adaptive_federated_optimization python main.py --config-path conf/cifar10 strategy=fedyogi
Traceback (most recent call last):
  File "main.py", line 5, in <module>
    import flwr as fl
ModuleNotFoundError: No module named 'flwr'
```

Reply:
It was probably the commented requirement you spotted earlier. I ran it again and it worked. Feel free to try again if you want

Reply:
I think the readme should be updated to indicate that once should do first `poetry shell` and then run the python command you have. I'd say lets assume nobody knows anything about virtualenvs or poetry.

Reply:
also, the `strategy=fedyogi` doesn't seem to work (it runs but hits `NaN` loss almost immediately`) so maybe let's put good old `FedAvg` as the first example line?

Reply:
Are you using version 1.44 of `grpcio` ? This might be related to: https://github.com/ray-project/ray/issues/22518

Reply:
it seems `poetry install` now does pull `grpcio 1.44`. I ran the example and seems to be fine (also w/ FedYogi)

Reply:
But i still stand to my previous message that it would be good to have some very minor instructions on how to setup the environment. If someone just comes to the readme of this example and tries to run the command in the readme, only error messages await. I'd say is enough to add a link to the top baselines readme (showing how to setup the env) and then update the readme here and add `poetry run python3 main.py<...>` instead of directly doing `python main.py <..>`